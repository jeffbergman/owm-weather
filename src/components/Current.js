import React, { Component } from 'react';
import { titleCase, iconNbr } from '../helpers';

class Current extends Component {
	//displays the current weather data

	render() {

		let temp; 
		let desc;
		let iconClass;
		let id;
		let icon;

		//the else part should only happen the first time
		//when there's no data in localstorage, or if the 
		//user is in mobile sarafi in private mode
		if (this.props.current) {
			temp = Math.round(this.props.current.main.temp);
			id = this.props.current.weather[0].id;
			desc= titleCase(this.props.current.weather[0].description);
			icon = this.props.current.weather[0].icon;
			iconClass = iconNbr(id, icon);
		} else {
			temp = 0;
			desc= 'Loading data...';
			iconClass = 'owf-200';
		}

	  return (
    	<div>
	      <div className="icon-cont">
	        <i className={`owf ${iconClass} owf-8x`}></i>
	      </div>

	      <div className="weather-desc">
	        {desc}  
	      </div>
	      
	      <div className="temp-cont">
	        <span className="temp-nbr">{temp}</span>
	        <span className="temp-symbol">&deg;</span>
	      </div>
      </div>
    );
  } 
}

export default Current;