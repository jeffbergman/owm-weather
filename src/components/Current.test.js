import React from 'react';
import { shallow } from 'enzyme';
import Current from './Current';

describe('Current', () => {
	
	let props = {};
	let current = shallow(<Current {...props} />);

	it('renders correctly', () => {
		expect('current').toMatchSnapshot();
	});

	describe('When !props.current', () => {

		it('displays a temperature of 0', () => {
			expect(current.find('.temp-nbr').text()).toEqual('0');
		});

		it('displays a Loading message', () => {
			expect(current.find('.weather-desc').text()).toEqual('Loading data...');
		});

		it('displays a default icon of owf-200', () => {
			expect(current.find('.owf-200').exists()).toBe(true);
		});
	});

	describe('When there are valid current props', () => {

		beforeEach(() => {
      props = { 
      	current: { 
	        "weather":[{"id":800,"main":"Clear","description":"clear sky","icon":"01d"}],
	        "main": {
	          "temp":65.84,
	          "pressure":1019,
	          "humidity":43,
	          "temp_min":59,
	          "temp_max":69.8
	        },
	        "dt":1522346340,
	        "sunrise":1522331019,
	        "sunset":1522375970
	      }  
      };
      current = shallow(<Current {...props} />);
    });

		it('displays a temperature of 66 when main.temp is 65.84', () => {
			expect(current.find('.temp-nbr').text()).toEqual('66');
		});


		it('displays a description "Clear Sky" weather[0].description is "clear sky"', () => {
			expect(current.find('.weather-desc').text()).toEqual('Clear Sky');
		});

		it('displays an icon of owf-800-d when weather[0].id is 800 and weather.icon is 01d', () => {
			expect(current.find('.owf-800-d').exists()).toBe(true);
		});

	});

});