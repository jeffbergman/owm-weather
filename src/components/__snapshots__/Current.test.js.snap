// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current renders correctly 1`] = `"current"`;
